desc/compute/data_index.py

```diff
@@ -98,14 +99,16 @@ def register_compute_fun(
         a flux function, etc.
     data : list of str
         Names of other items in the data index needed to compute qty.
-    parameterization: str or list of str
-        Name of desc types the method is valid for. eg 'desc.geometry.FourierXYZCurve'
-        or `desc.equilibrium.Equilibrium`.
-    axis_limit_data : list of str
-        Names of other items in the data index needed to compute axis limit of qty.
     aliases : list
         Aliases of `name`. Will be stored in the data dictionary as a copy of `name`s
         data.
+    parameterization : str or list of str
+        Name of desc types the method is valid for. eg `'desc.geometry.FourierXYZCurve'`
+        or `'desc.equilibrium.Equilibrium'`.
+    grid_type : str
```

Comment:
The latter; it's for the logic in `Equilibrium.compute` to ensure intermediate quantities are computed on their proper grid. We currently check this based on the data index fields `dim` and `coordinates`, for example we make all `dim=0` quantities computed on a quadrature grid. But this setup is messy and won't work well for all compute quantities (for example this updated elongation). 

The idea of adding the `grid_type` is to explicitly label which grid type should be used instead of basing it off of the dimension and coordinates used. I start the process here because I needed it, but we will have to update all the other compute quantities in a future PR.